src/detect-replace.c

```diff
@@ -276,15 +276,11 @@ int DetectReplaceLongPatternMatchTest(uint8_t *raw_eth_pkt, uint16_t pktsize,
     DecodeEthernet(&th_v, &dtv, p, GET_PKT_DATA(p), pktsize);
 
     DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(de_ctx);
     de_ctx->flags |= DE_QUIET;
 
     de_ctx->sig_list = SigInit(de_ctx, sig);
-    if (de_ctx->sig_list == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(DetectEngineAppendSig(de_ctx, sig));
```

Comment:
Is it right to use the DetectEngineAppendSig function here?

Reply:
From my understanding, it would be, yes.

Reply:
`DetectEngineAppendSig` is preferred, but needs to be used instead of `SigInit`